rootex/framework/components/game/spring_arm_component.cpp

```diff
@@ -0,0 +1,45 @@
+#include "spring_arm_component.h"
+
+DEFINE_COMPONENT(SpringArmComponent);
+
+SpringArmComponent::SpringArmComponent(Entity& owner, const JSON::json& data)
+    : Component(owner)
+    , m_DesiredLocalPosition(data.value("desiredlocalposition", Vector3::Zero))
+    , m_Lerp(data.value("lerp", 0.0f))
+    , m_DependencyOnTransformComponent(this)
+{
+}
+
+JSON::json SpringArmComponent::getJSON() const
+{
+	JSON::json j;
+
+	j["lerp"] = m_Lerp;
+	j["desiredlocalposition"] = m_DesiredLocalPosition;
+
+	return j;
+}
+
+bool SpringArmComponent::setupData()
+{
+	m_InitialPosition = m_DependencyOnTransformComponent.getComponent()->getParentAbsoluteTransform().Translation() + m_DependencyOnTransformComponent.getComponent()->getPosition();
```

Comment:
We can just use `getTransformComponent()`. Using the dependency object is really cumbersome here.